File: 01.JS-Basic/По сложни проверки Упражнение/OnTimeForTheExam.js
function onTime(input) {

    let examHour = Number(input[0])
    let examMinute = Number(input[1])
    let hourArrive = Number(input[2])
    let minuteArrive = Number(input[3])

    let examInMinutes = examHour * 60 + examMinute
    let arriveInMinets = hourArrive * 60 + minuteArrive

    if (examInMinutes === arriveInMinets) {
        console.log("On time")
    }
    else if ((examInMinutes - arriveInMinets) <= 30 && (examInMinutes - arriveInMinets) >= 0) {
        console.log("On time")
        console.log(`${examInMinutes - arriveInMinets} minutes before the start`)
    }
    else if ((examInMinutes - arriveInMinets) < 60 && (examInMinutes - arriveInMinets) >= 0) {
        console.log("Early")
        console.log(`${examInMinutes - arriveInMinets} minutes before the start`)
    }
    else if ((examInMinutes - arriveInMinets) >= 60) {
        if ((examInMinutes - arriveInMinets) % 60 < 10) {
            console.log("Early")
            console.log(`${Math.floor((examInMinutes - arriveInMinets) / 60)} : 0${(examInMinutes - arriveInMinets) % 60} hours before the start`)
        }
        else {
            console.log("Early")
            console.log(`${Math.floor((examInMinutes - arriveInMinets) / 60)} : ${(examInMinutes - arriveInMinets) % 60} hours before the start`)
        }
    }
    else if (arriveInMinets < examInMinutes + 60) {
        console.log("Late")
        console.log(`${arriveInMinets - examInMinutes} minutes after the start`)
    }
    else if ((arriveInMinets - examInMinutes) >= 60) {
        if ((arriveInMinets - examInMinutes) % 60 < 10) {
            console.log("Late")
            console.log(`${Math.floor((arriveInMinets - examInMinutes) / 60)} : 0${(arriveInMinets - examInMinutes) % 60} hours after the start`)
        }
        else {
            console.log("Late")
            console.log(`${Math.floor((arriveInMinets - examInMinutes) / 60)} : ${(arriveInMinets - examInMinutes) % 60} hours after the start`)
        }
    }
}
onTime(["14",
    "00",
    "13",
    "55"])

File: JAVA-Basic/JavaBasic/src/ConditionalStatementsAdvanced/Exercise/OnTimeForTheExam.java
package ConditionalStatementsAdvanced.Exercise;

import java.util.Scanner;

public class OnTimeForTheExam {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int examH = Integer.parseInt(scan.nextLine());
        int examM = Integer.parseInt(scan.nextLine());
        int arriveH = Integer.parseInt(scan.nextLine());
        int arriveM = Integer.parseInt(scan.nextLine());

        int examTimeInMins = examH * 60 + examM;
        int arriveTimeInMins = arriveH * 60 + arriveM;
        int timeDifference = Math.abs(examTimeInMins - arriveTimeInMins);

        if (arriveTimeInMins > examTimeInMins) {
            System.out.println("Late");
            if (timeDifference < 60) {
                System.out.printf("%d minutes after the start", timeDifference);
            } else {
                if (timeDifference % 60 >= 10) {
                    System.out.printf("%d:%d hours after the start", timeDifference / 60, timeDifference % 60);
                } else {
                    System.out.printf("%d:0%d hours after the start", timeDifference / 60, timeDifference % 60);
                }
            }
        } else if (arriveTimeInMins >= examTimeInMins - 30) {
            if (arriveTimeInMins == examTimeInMins) {
                System.out.println("On time");
            } else {
                System.out.println("On time");
                System.out.printf("%d minutes before the start", timeDifference);
            }
        } else {
            System.out.println("Early");
            if (timeDifference < 60) {
                System.out.printf("%d minutes before the start", timeDifference);
            } else {
                if (timeDifference % 60 >= 10) {
                    System.out.printf("%d:%d hours before the start", timeDifference / 60, timeDifference % 60);
                } else {
                    System.out.printf("%d:0%d hours before the start", timeDifference / 60, timeDifference % 60);
                }
            }
        }
    }
}
